spec/unit/resource_spec.rb

```diff
@@ -492,6 +492,20 @@ def self.name
       expect(r.resource_name).to eq :blah
       expect(r.declared_type).to eq :d
     end
+
+    # This tests some somewhat confusing behavior that used to occur due to the resource_name call
```

Comment:
I had to read this test... a lot... to understand what you were trying to test. More comments here on what the test is testing would be useful.